app/controllers/api/v1/graphql_controller.rb

```diff
@@ -149,7 +149,7 @@ def format_error_message(e)
       private
 
       def authenticate_graphql_user
-        params[:query].to_s.match(/^((query )|(mutation[^\{#]*{\s*(resetPassword|changePassword|resendConfirmation|userDisconnectLoginAccount)\())/).nil? ? authenticate_user! : authenticate_user
+        !!(params[:query].to_s.match(/^(query\s+Me\s*\{|(mutation[^{#]*{\s*(resetPassword|changePassword|resendConfirmation|userDisconnectLoginAccount)\())/)) ? authenticate_user : authenticate_user!
```

Comment:
Suggestion:
- We could use `match?` since we want to know if it is true or false anyway
- I think we could make this a bit easier to read if we broke it down a bit more. Maybe by adding a `safe_mutations` array, and interpolating it? Maybe, something like this (but we could break it down even more if we think it is worth it):

```
      SAFE_MUTATIONS = %w[
        resetPassword
        changePassword
        resendConfirmation
        userDisconnectLoginAccount
      ]

      def authenticate_graphql_user
        params[:query].to_s.match?(/^(query\s+Me\s*\{|(mutation[^{#]*{\s*(#{SAFE_MUTATIONS.join("|")})\())/) ?
        authenticate_user : authenticate_user!
      end
```

What do you think, @melsawy?

Reply:
@vasconsaurus I rewrote the method because I discovered another way to obtain the type (mutation or query) and the name, as Caio mentioned.